packages/compiler-core/src/transforms/vSlot.ts

```diff
@@ -49,6 +49,19 @@ export const trackSlotScopes: NodeTransform = (node, context) => {
     (node.tagType === ElementTypes.COMPONENT ||
       node.tagType === ElementTypes.TEMPLATE)
   ) {
+    const { parent } = context
```

Comment:
It’s not quite appropriate to add this logic to `trackSlotScopes` as it doesn’t align with the SRP(Single Responsibility Principle). I still think it’s better to place the validation in the parser (as in your first submitted version), which is also how Vue2 seems to handle it during the parse phase.

Reply:
thanks for the feedback :D will change that later